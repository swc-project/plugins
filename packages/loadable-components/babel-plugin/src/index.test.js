/* eslint-disable import/no-extraneous-dependencies */
import { transform } from '@babel/core'
import plugin from '.'

const testPlugin = code => {
  const result = transform(code, {
    plugins: [plugin],
    configFile: false,
  })

  return result.code
}

describe('plugin', () => {
  describe('Magic comment', () => {
    it('should transpile shortand properties', () => {
      const result = testPlugin(`
        const obj = {
          /* #__LOADABLE__ */
          load() {
            return import('moment')
          }
        }
      `)

      expect(result).toMatchSnapshot()
    })

    it('should transpile arrow functions', () => {
      const result = testPlugin(`
        const load = /* #__LOADABLE__ */ () => import('moment')
      `)

      expect(result).toMatchSnapshot()
    })

  })
})
