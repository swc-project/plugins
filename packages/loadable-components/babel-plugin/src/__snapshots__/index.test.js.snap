// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin Magic comment should transpile arrow functions 1`] = `
"const load = {
  resolved: {},

  chunkName() {
    return \\"moment\\";
  },

  isReady(props) {
    const key = this.resolve(props);

    if (this.resolved[key] !== true) {
      return false;
    }

    if (typeof __webpack_modules__ !== 'undefined') {
      return !!__webpack_modules__[key];
    }

    return false;
  },

  importAsync: () => import(
  /* webpackChunkName: \\"moment\\" */
  'moment'),

  requireAsync(props) {
    const key = this.resolve(props);
    this.resolved[key] = false;
    return this.importAsync(props).then(resolved => {
      this.resolved[key] = true;
      return resolved;
    });
  },

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ !== 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"moment\\");
    }

    return eval('require.resolve')(\\"moment\\");
  }

};"
`;

exports[`plugin Magic comment should transpile shortand properties 1`] = `
"const obj = {
  load: {
    resolved: {},

    chunkName() {
      return \\"moment\\";
    },

    isReady(props) {
      const key = this.resolve(props);

      if (this.resolved[key] !== true) {
        return false;
      }

      if (typeof __webpack_modules__ !== 'undefined') {
        return !!__webpack_modules__[key];
      }

      return false;
    },

    importAsync: () => {
      return import(
      /* webpackChunkName: \\"moment\\" */
      'moment');
    },

    requireAsync(props) {
      const key = this.resolve(props);
      this.resolved[key] = false;
      return this.importAsync(props).then(resolved => {
        this.resolved[key] = true;
        return resolved;
      });
    },

    requireSync(props) {
      const id = this.resolve(props);

      if (typeof __webpack_require__ !== 'undefined') {
        return __webpack_require__(id);
      }

      return eval('module.require')(id);
    },

    resolve() {
      if (require.resolveWeak) {
        return require.resolveWeak(\\"moment\\");
      }

      return eval('require.resolve')(\\"moment\\");
    }

  }
};"
`;